### DISCUSSION TITLE
RTS/CTS configuration with RS-485 server

### DISCUSSION THREAD

**Discussion by StewartInnes** [pymodbus-dev/pymodbus#2012] [Help] [answered]
I'm creating a server with half-duplex RS-485 communications, so I need to enable RTS/CTS (really just RTS in my case) but I have a hardware bug that requires me to change RTS to active-high. In PySerial, I can use 
```
serial.rs485.RS485Settings(
  rts_level_for_tx = False,
  rts_level_for_rx = True
)
```

to change the RTS logic. 

How would you do this with pymodbus?

I did find this: https://github.com/pymodbus-dev/pymodbus/issues/33#issuecomment-939328396
But it's over 10 years old so I want to see if there's some built-in dev that I'm missing.

**Comment by janiversen** [comment, marked as answer]
I have no idea, it's not something you configure.

both the sync and async client uses pyserial so you might be able to patch either pyserial or where pymodbus starts pyserial.

**Reply by StewartInnes** [reply]
Thanks for the quick response! That confirmation that it's not configurable is what I needed.
